include/hipSYCL/sycl/libkernel/detail/vec.hpp

```diff
@@ -78,20 +78,82 @@ template<> struct logical_vector_op_result<detail::dp_float>
 
 template <int...> struct vector_index_sequence {};
 
+namespace emulated_types {
+
+using uchar = unsigned char;
+using ushort = unsigned short;
+using uint = unsigned int;
+using longlong = long long;
+using ulong = unsigned long;
+using ulonglong = unsigned long long;
+
+template <class T, int Dim> struct required_alignment {
+#if defined(HIPSYCL_PLATFORM_CUDA)
+  constexpr static size_t value = alignof(T) * Dim;
+#else
+  // Use default alignment otherwise
+  constexpr static size_t value = 0;
+#endif
+};
+
+// TODO: This should probably be moved to a backend-specific location.
+#if defined(HIPSYCL_PLATFORM_CUDA)
```

Comment:
Wouldn't it suffice to only enable this in the host pass of CUDA/HIP backends? e.g. `defined(HIPSYCL_PLATFORM_CPU) && defined(HIPSYCL_PLATFORM_ROCM/CUDA)`.